master/buildbot/reporters/github.py

```diff
@@ -81,7 +81,8 @@ def createStatus(self,
             [self.baseURL, 'repos', repo_user, repo_name, 'statuses', sha]), json=payload)
 
     @defer.inlineCallbacks
-    def send(self, build):
+    def send(self, postData):
```

Comment:
Hm. I think I was holding "send" as a method that purely handled http requests for the data it was given (how I tried to hold the hipchat reporter).

It seems that the interface (in all cases) actually just expects a build and then does some filtering/logic to end up with the final json payload.

I can follow that pattern in my work as well and remove the need to modify this interface.